posthog/hogql_queries/insights/trends/data_warehouse_trends_query_builder.py

```diff
@@ -0,0 +1,526 @@
+from typing import List, Optional, cast
+from posthog.hogql import ast
+from posthog.hogql.parser import parse_expr, parse_select
+from posthog.hogql.property import property_to_expr
+from posthog.hogql.timings import HogQLTimings
+from posthog.hogql_queries.insights.trends.aggregation_operations import (
+    AggregationOperations,
+)
+from posthog.hogql_queries.insights.trends.breakdown import Breakdown
+from posthog.hogql_queries.insights.trends.display import TrendsDisplay
+from posthog.hogql_queries.utils.query_date_range import QueryDateRange
+from posthog.models.filters.mixins.utils import cached_property
+from posthog.models.team.team import Team
+from posthog.schema import HogQLQueryModifiers, TrendsQuery, DataWarehouseNode
+
+
+class DataWarehouseTrendsQueryBuilder:
```

Comment:
We may wanna add a base abstract class to keep the function signatures consistent between this and the `TrendsQuerybuilder`